tests/java/verifiers/src/main/java/io/vectorized/reads_writes/Workload.java

```diff
@@ -0,0 +1,437 @@
+      started_s = System.currentTimeMillis() / 1000;
+      System.out.println(
+          "#workload\tlogical_time\tseconds_since_start\tpartition\tlog_type\tdetails");
+    }
+    log(-1, "start\t" + args.name);
+
+    write_threads = new ArrayList<>();
+    read_threads = new ArrayList<>();
+
+    oids = new HashMap<>();
+    ops = new HashMap<>();
+    rsems = new HashMap<>();
+    wstat = new HashMap<>();
+
+    for (int partition = 0; partition < args.partitions; partition++) {
+      oids.put(partition, new LinkedList<>());
+      rsems.put(partition, new Semaphore(0));
+      wstat.put(partition, new WriterStat());
+
+      final int p = partition;
+      write_threads.add(new Thread(() -> {
+        try {
+          writeProcess(p);
+        } catch (Exception e) {
+          synchronized (this) {
+            System.out.println("=== write process error");
+            System.out.println(e);
+            e.printStackTrace(System.out);
+            System.exit(1);
+          }
+        }
+      }));
+
+      read_threads.add(new Thread(() -> {
+        try {
+          readProcess(p);
+        } catch (Exception e) {
+          synchronized (this) {
+            System.out.println("=== read process error");
+            System.out.println(e);
+            e.printStackTrace(System.out);
+            System.exit(1);
+          }
+        }
+      }));
+    }
+
+    for (var th : write_threads) {
+      th.start();
+    }
+
+    for (var th : read_threads) {
+      th.start();
+    }
+  }
+
+  public void stop() throws Exception {
+    is_active = false;
+
+    log(-1, "stop requested");
+
+    synchronized (this) {
+      for (int partition = 0; partition < args.partitions; partition++) {
+        rsems.get(partition).release();
+      }
+    }
+  }
+
+  public void waitStopped() throws Exception {
+    if (is_active) {
+      throw new Exception("wasn't asked to stop");
+    }
+    if (are_readers_stopped) {
+      return;
+    }
+
+    for (var th : write_threads) {
+      th.join();
+    }
+    log(-1, "writers stopped");
+    are_writers_stopped = true;
+    for (var th : read_threads) {
+      th.join();
+    }
+    log(-1, "readers stopped");
+    are_readers_stopped = true;
+  }
+
+  private void writeProcess(int partition) throws Exception {
+    Properties props = new Properties();
+    props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, args.brokers);
+    props.put(ProducerConfig.ACKS_CONFIG, "all");
+    props.put(
+        ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG,
+        "org.apache.kafka.common.serialization.StringSerializer");
+    props.put(
+        ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG,
+        "org.apache.kafka.common.serialization.StringSerializer");
+    props.put(ProducerConfig.LINGER_MS_CONFIG, 0);
+    props.put(ProducerConfig.MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION, 5);
+    props.put(ProducerConfig.RETRIES_CONFIG, Integer.MAX_VALUE);
+    props.put(ProducerConfig.ENABLE_IDEMPOTENCE_CONFIG, true);
```

Comment:
Would it be possible to make this configurable ?

Reply:
I don't like adding configuration before we have a use case for that, it just extra work with an uncertain benefit. If we need it we'll add it later